### PR TITLE
Create table of content with anchors to ease navigation within views 

### PR DESCRIPTION
**Pull request by TheHelias**
https://www.loom.com/share/f818f5c9c90b4ee49a6b81428521211d

### CODE REVIEW

File: src/lib/SectionTitle.svelte [hunk 1]
```diff
@@ -1,13 +1,30 @@
...
   }
+  const context = getContext("TableOfContents")
+  onMount(() => {
+    if (context) {
+      if (title && id) {
+        context.addEntry({
+          title: title,
+          level: level,
+          id: id,
+          active: $tableOfContents.length > 0 ? false : true,
```
Comment (lietu) [inline src/lib/SectionTitle.svelte:20]: This seems wrong and confusing, and why is the children accessing the store? If any logic like this needs to exist it should be in the parent.

File: src/lib/TableOfContents.svelte [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  <Grid sm={$isSmallScreen ? 10 : 9}>
+    <slot />
+  </Grid>
+  {#if !$isSmallScreen}
+    <Grid sm={2} class="table-of-contents">
+      <div class="content">
+        <p>On this page</p>
+        {#each $tableOfContents as entry}
+          <a
+            on:click={updateActiveEntry(entry.id)}
```
Comment (lietu) [inline src/lib/TableOfContents.svelte:57]: If a table of contents has any "active" styles it definitely should not be based on what you've clicked on last, but which section you're scrolled to. I don't think this is a useful feature in general to have and I would rather not waste time on figuring out the scroll handlers for when the items are set active

File: src/lib/TableOfContents.svelte [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            class={`links ${entry.active ? "active" : ""}`}
```
Comment (lietu) [inline src/lib/TableOfContents.svelte:58]: The class should be `link` not `links` as this does not contain multiple links

File: src/lib/TableOfContents.svelte [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+    ::-webkit-scrollbar-thumb {
+      background: $color-primary-dark-hover;
+    }
+
+    .content {
+      display: flex;
+      flex-direction: column;
+      position: -webkit-sticky;
+      position: sticky;
```
Comment (lietu) [inline src/lib/TableOfContents.svelte:105]: The lower one overrides the one above

File: src/lib/TableOfContents.svelte [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+        padding-right: 1rem;
+      }
+      .links {
+        @include text-body-2();
+        color: $color-neutral-gray;
+        text-decoration: none;
+        padding: 0.5rem 0 0.5rem 1rem;
+        margin-left: 0.5rem;
+        border-left: 2px solid $color-dark-green;
+        transition: color 0.3s ease, border-left-color 0.3s ease;
```
Comment (lietu) [inline src/lib/TableOfContents.svelte:131]: `all`? is there any property you're changing that you do not want to transition?

File: src/lib/components/Hamburger.svelte [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  export let popover: any
+  export let showInTablet: boolean = false
+  export let alignIconToRight: boolean = false
+</script>
+
+<button
+  use:popover.button
+  class={`hamburger-button ${showInTablet ? "tablet" : ""} ${
+    alignIconToRight ? "align-right" : ""
+  }`}
```
Comment (lietu) [inline src/lib/components/Hamburger.svelte:14]: https://learn.svelte.dev/tutorial/classes

File: src/lib/components/Hamburger.svelte [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+    &.tablet {
+      @include tablet() {
+        display: block;
+      }
+    }
+
+    &.align-right {
+      float: right;
+    }
```
Comment (lietu) [inline src/lib/components/Hamburger.svelte:44]: I'd prefer basically any other way of aligning this than `float`, it tends to cause a lot of issues

File: src/lib/images/close-icon.svg
Comment (lietu) [inline src/lib/images/close-icon.svg:1]: To me it seems wrong that these icons contain borders

File: src/lib/TableOfContents.svelte (hunk 2 above)
Comment (TheHelias) [reply src/lib/TableOfContents.svelte:57]: Active styles have been removed pending the time we resolve what it should look like and how they should work

**Review by fbjorn — APPROVED** [review]

**Pull request merged**